packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -3112,6 +3165,45 @@ export function captureCommitPhaseError(sourceFiber: Fiber, error: mixed) {
   }
 }
 
+export function captureCommitPhaseErrorForUnmountedFiber(
+  sourceFiber: Fiber,
+  nearestMountedAncestor: Fiber,
+  error: mixed,
+) {
+  let fiber = nearestMountedAncestor.return;
```

Comment:
The nearestMountedAncestor is by definition a parent of the thing that threw. So why are we starting by checking one level above that? It seems to me that this would target the wrong ancestor if the thing being deleted is a direct child of the error boundary itself. Your tests doesn't seem to test this case because they all use the `Conditional` wrapper.

Reply:
> The nearestMountedAncestor is by definition a parent of the thing that threw.

No. The thing that threw may be further down inside of the unmounted tree. It doesn't necessarily have to be the root of the unmounted tree.

> So why are we starting by checking one level above that?

We've already disconnected the `return` pointer after layout. (This is the mechanism we rely on for detecting and ignoring (and warning about) an unmounted state update.

Reply:
Oh, I think I get what you're saying. You're sound the "nearest mounted ancestor" is *already* an ancestor, so starting with `.return` is wrong. I gotcha. The "parent" comment threw me off.